source/adapters/hip/device.cpp

```diff
@@ -724,11 +724,20 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
   }
 
   case UR_DEVICE_INFO_GLOBAL_MEM_FREE: {
+    // Work around an issue on some (unsupported) architectures,
+    // where hipMemGetInfo fails internally and returns hipErrorInvalidValue
+    // when trying to query the amount of available global memory. Since we
+    // can't distinguish this condition from us doing something wrong, we can't
+    // handle it gracefully.
+    hipDeviceProp_t Props;
+    UR_CHECK_ERROR(hipGetDeviceProperties(&Props, hDevice->get()));
+    if (strcmp(Props.gcnArchName, "gfx1031") == 0) {
+      return ReturnValue(0);
```

Comment:
You mention `hipErrorInvalidValue` in the comment above, but we're not checking for it here?

Reply:
Yeah, maybe I am not being clear, sorry. I'm trying to say that `hipMemGetInfo` below returns `hipErrorInvalidValue` if we let that run, and we just want to avoid that because it's ambiguous as to whether it means we're passing incorrect values, or it just internally can't do what it needs to do. Instead of falling into that situation, we just early-return a dummy value here.

Reply:
Can we instead return the dummy value here for gfx1031 _only if_ the result code of `hipMemGetInfo` is `hipErrorInvalidValue`, then call `checkErrorUR` on the result?

IIRC, we haven't always had that error. However, I may be very wrong, and if it is always an issue for all relevant ROCm versions, then I guess no need the extra complexity to this workaround, so I am happy leaving it as is.